service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ManageOrdersMidEventControllerTest.java

```diff
@@ -215,11 +232,123 @@ void reviewShouldNotAlterCaseData() {
         assertThat(response.getData()).isEqualTo(caseDetails.getData());
     }
 
+    @Test
+    void epoEndDateShouldReturnErrorForPastDate() {
+        CaseData caseData = buildCaseData().toBuilder().manageOrdersEventData(
+            buildRemoveToAccommodationEventData(now().plusDays(1), now().minusDays(1))).build();
+
+        AboutToStartOrSubmitCallbackResponse response = postMidEvent(caseData, "order-details");
+
+        assertThat(response.getErrors()).containsOnly("Enter an end date in the future");
+    }
+
+    @Test
+    void epoEndDateShouldReturnErrorWhenEndDateIsNotInRangeWithApprovalDate() {
+        final LocalDateTime approvalDate = LocalDateTime.now().minusDays(5);
+
+        CaseData caseData = buildCaseData().toBuilder().manageOrdersEventData(
+            buildRemoveToAccommodationEventData(approvalDate, approvalDate.plusDays(9))).build();
+
+        AboutToStartOrSubmitCallbackResponse response = postMidEvent(caseData, "order-details");
+
+        assertThat(response.getErrors()).containsOnly("Emergency protection orders cannot last longer than 8 days");
+    }
```

Comment:
small point but by being more strict about the boundary of the date range, e.g. `1 day plus 1 second`,  you could improve the strength of this test, which would mean even slight regressions would be caught, like 
```
        CaseData caseData = buildCaseData().toBuilder().manageOrdersEventData(
            buildRemoveToAccommodationEventData(approvalDate, approvalDate.plusDays(8).plusSeconds(1))).build();
```